src/utils_request.js

```diff
@@ -121,7 +119,6 @@ export const requestAsBrowser = async (options = {}) => {
         headerGeneratorOptions: ow.optional.object,
         stream: ow.optional.boolean,
         decodeBody: ow.optional.boolean,
-        forceUrlEncoding: ow.optional.boolean,
```

Comment:
although very new addition, removing it seems like a BC to me, so should be at least noted in the changelog

i am also not very convinced by the `fixUrl` function, we should still allow control over how this behaves, even if we want to use this kind of autofix - as it does not work in 100% cases (e.g. the `%cf` example - or did i miss something?)

Reply:
i am just concerned with the original problem we wanted to address and basically decided not to change the url automatically in the end (https://apifier.slack.com/archives/CGZSN9DQC/p1619201727049400)

i believe in that particular site the URL can contain things like `%cf` in the query string and the change in tests indicates such case would be now impossible to handle as we would "auto-fix" to wrong symbol instead of keeping it intact

(granted, the problematic actor was not using `got` at all originally, but i was able to reproduce it locally with sdk 1.3 or something with `got-scraping` too)

Reply:
Yeah, we were waiting for your insights as well. You can discuss the `%cf` specifically with @szmarczak, but he said that it's not valid utf-8. So we made the call that if we can autofix everything except invalid chars, then it's ok. But I did not know what kind of thoughts you put into this so I did not want to merge this before you get back.

Reply:
@B4nan Thanks for feedback! Got always throws on invalid URL. It's [normalized](https://github.com/sindresorhus/got/blob/79547b35067f141cd6c1c528e5042fa4662fdf97/source/core/options.ts#L1282-L1284) via `new URL(url)` followed by `decodeURI(url)`. We deeply care about stability so I doubt this is going to be removed. 

One workaround would be to store the proper URL in a context and then replace it in an `Agent.addRequest`. This way invalid URLs would be accepted. An disadvantage is that Got still needs to receive a valid URL somehow - could be via the `fixUrl` function.

Reply:
Yeah I actually saw that `decodeURI` call (that ignored the return value) and realized it is meant as a validation. 

It still feels like a BC to me, we had a working flag, now it is no longer there and we need to call `encodeURI` explicitly, as the `fixUrl` function will not handle the `%cf` case correctly. Not saying we should ship v3 because of this, but it should be noted in the changelog as a possible BC or side effect.

I hope it makes some sense, I am getting more and more confused looking into this :D

Reply:
FYI: the `decodeURI` firstly appeared in https://github.com/sindresorhus/got/pull/441

> we had a working flag

There were some issues it. E.g. double encoding `https://example.com/%25` would be incorrectly converted into `https://example.com/%2525`, which the server would interpret as `https://example.com/%25` instead of `https://example.com/%`. People would still need to properly encode it at first try with the flag being `false`.

> will not handle the %cf case correctly

If `correctly` means against the spec, then yes. Although `fixUrl` is compliant with the spec. `%cf` is not a valid percent encoded character so it'll be rewritten as utf-8 replacement character. `%xx` is definitely not utf-8 so it's rewritten as `%25xx`. It's the closest behavior we can get without tinkering with the internals.

Reply:
> There were some issues it. E.g. double encoding

That is why the flag was opt-in only. 

Anyway, not trying to argue what is better approach. Just pointing out that the behaviour is changed, so should be noted in changelog. 

> If correctly means against the spec

This is not about specs. We had a flag to automatically encoded the URLs. The flag is not there anymore, if anyone was using it, the actor will break in cases like the `%cf`. 

But again, it is very recent addition, and there is a good chance nobody is using it.

Reply:
> Anyway, not trying to argue what is better approach. Just pointing out that the behaviour is changed, so should be noted in changelog.

Me neither. I know it would be best to support `%cf` as it is but currently it's out of our scope (unless we do an `Agent` patch). It depends on how important this issue is.

> This is not about specs. We had a flag to automatically encoded the URLs. The flag is not there anymore, if anyone was using it, the actor will break in cases like the %cf.

Indeed, this is a breaking change. What do you suggest? Should I restore `forceUrl`?

Reply:
that's what major versions are for 🔨

Reply:
we could also throw some meaningful error if so the `forceUrlEncoding` is used, safer than expecting people to read the release notes. as before it would end up with error, but now the `fixUrl` would handle it somehow and we could end up with silent failures.

personally i'd just keep it there, it does not hurt anyone and we can mark it as deprecated and remove it in v3. i know it is an edge case, and a recent addition that might not be really used, but it is still a BC and those should indeed happen in major bumps.